internal/xds/rbac/converter.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rbac
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	v1xdsudpatypepb "github.com/cncf/xds/go/udpa/type/v1"
+	v3xdsxdstypepb "github.com/cncf/xds/go/xds/type/v3"
+	v3rbacpb "github.com/envoyproxy/go-control-plane/envoy/config/rbac/v3"
+	"google.golang.org/grpc/authz/audit"
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func buildLogger(loggerConfig *v3rbacpb.RBAC_AuditLoggingOptions_AuditLoggerConfig) (audit.Logger, error) {
+	if loggerConfig.AuditLogger.TypedConfig == nil {
+		return nil, fmt.Errorf("AuditLogger TypedConfig cannot be nil")
+	}
+	customConfig, loggerName, err := getCustomConfig(loggerConfig.AuditLogger.TypedConfig)
+	if err != nil {
+		return nil, err
+	}
+	if loggerName == "" {
+		return nil, fmt.Errorf("AuditLogger TypedConfig.TypeURL cannot be an empty string")
+	}
+	factory := audit.GetLoggerBuilder(loggerName)
+	if factory == nil {
+		if loggerConfig.IsOptional {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("no builder registered for %v", loggerName)
+	}
+	auditLoggerConfig, err := factory.ParseLoggerConfig(customConfig)
+	if err != nil {
+		return nil, fmt.Errorf("AuditLogger custom config could not be parsed: %v", err)
+	}
+	auditLogger := factory.Build(auditLoggerConfig)
+	return auditLogger, nil
+}
+
+func getCustomConfig(config *anypb.Any) (json.RawMessage, string, error) {
+	switch config.GetTypeUrl() {
+	case "type.googleapis.com/udpa.type.v1.TypedStruct":
```

Comment:
This structure is copied from the `lb` policy (https://github.com/grpc/grpc-go/blob/ccad7b7570fd233e4558b034034672c0de89c87c/xds/internal/xdsclient/xdslbregistry/converter.go#L71-L94)

I suspect if there were already consts they would've gotten used here. But we can certainly define consts. The naming is a potentially sticky point
The import for this package is, canonically in this repo, `v1xdsudpatypepb` and `v3xdsxdstypepb`

Could we do the consts as this?
```
const udpaTypedStuctType = "type.googleapis.com/udpa.type.v1.TypedStruct"
const xdsTypedStuctType = "type.googleapis.com/xds.type.v3.TypedStruct"

```

Reply:
> This code-structure is copied from the lb policy (

This code has changed FWIW to be a registry instead of a switch statement.

I don't care a lot about the name of unexported consts as long as they aren't misleading or something.

Reply:
Will use the above const names

Reply:
PTAL